<launch>
  <param name="use_sim_time" value="true"/>

  <node name="stageros" pkg="stage_ros" type="stageros" args="$(find PA_rrts)/maps/engineering/eng_world.world"/>

  <node name="fake_localization_0" pkg="fake_localization" type="fake_localization">
    <remap from="/base_pose_ground_truth" to="/robot_0/base_pose_ground_truth"/>
    <remap from="/amcl_pose" to="/robot_0/amcl_pose"/>
    <remap from="/particlecloud" to="/robot_0/particlecloud"/>
    <param name="odom_frame_id" type="string" value="/robot_0/odom"/>
    <param name="base_frame_id" type="string" value="/robot_0/base_link"/>
<!--     <param name="global_frame_id" type="string" value="/map"/> -->
  </node>

  <node name="fake_localization_1" pkg="fake_localization" type="fake_localization">
    <remap from="/base_pose_ground_truth" to="/robot_1/base_pose_ground_truth"/>
    <remap from="/amcl_pose" to="/robot_1/amcl_pose"/>
    <remap from="/particlecloud" to="/robot_1/particlecloud"/>
    <param name="odom_frame_id" type="string" value="/robot_1/odom"/>
    <param name="base_frame_id" type="string" value="/robot_1/base_link"/>
  </node>

  <node name="fake_localization_2" pkg="fake_localization" type="fake_localization">
    <remap from="/base_pose_ground_truth" to="/robot_2/base_pose_ground_truth"/>
    <remap from="/amcl_pose" to="/robot_2/amcl_pose"/>
    <remap from="/particlecloud" to="/robot_2/particlecloud"/>
    <param name="odom_frame_id" type="string" value="/robot_2/odom"/>
    <param name="base_frame_id" type="string" value="/robot_2/base_link"/>
  </node>


  <node name="local_map" pkg="local_map" type="local_map_node">
    <remap from="scan_in" to="/robot_0/base_scan"/>
    <remap from="static_map" to="static_map_lane"/>
    <param name="robot_frame_id" type="string" value="/robot_0/base_link"/>
    <param name="map_w" type="double" value="5"/>
    <param name="map_h" type="double" value="5"/>
  </node>

  <node name="map_server_lane" pkg="map_server_ext" type="map_server" args="$(find PA_rrts)/maps/engineering/eng_lane_map.yaml">
    <remap from="static_map" to="static_map_lane"/>
    <remap from="map" to="lane_map"/>
  </node>
  
  <node name="golfcar_pp" pkg="golfcar_ppc" type="golfcar_pp" >
    <param name="coord_frame_id" type="string" value="/robot_0/odom"/>
    <param name="robot_frame_id" type="string" value="/robot_0/base_link"/>
    <remap from="/cmd_vel" to="/robot_0/cmd_vel"/>
  </node>

  <node name="speed_advisor" pkg="speed_advisor" type="speed_advisor">
    <remap from="/odom" to="/robot_0/odom"/>
    <param name="robot_frame_id" type="string" value="/robot_0/base_link"/>
    <remap from="/cmd_vel" to="/robot_0/cmd_vel"/>
  </node>

  <node name="rrts_node" pkg="rrts" type="rrts_node">
    <param name="robot_frame_id" type="string" value="/robot_0/base_link"/>
  </node>

  <node name="car3D" pkg="PA_rrts" type="car3D" >
    <param name="robot_frame_id" type="string" value="/robot_0/base_link"/>
  </node>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find rrts)/rvizconfig.rviz"/>

  <node name="car3D_0" pkg="PA_rrts" type="car3D">
    <param name="robot_frame_id" type="string" value="/robot_1/base_link"/>
    <param name="color" type="string" value="red"/>
  </node>

  <node name="box3D_1" pkg="PA_rrts" type="box3D" >
    <param name="obstacle_frame_id" type="string" value="/robot_2/base_link"/>
  </node>

</launch>
